layers/gpuav/instrumentation/gpuav_shader_instrumentor.cpp

```diff
@@ -1290,21 +1290,32 @@ static std::string LookupDebugUtilsNameNoLock(const DebugReport *debug_report, c
 
 // Generate the stage-specific part of the message.
 static void GenerateStageMessage(std::ostringstream &ss, uint32_t stage_id, uint32_t stage_info_0, uint32_t stage_info_1,
-                                 uint32_t stage_info_2, const std::vector<Instruction> &instructions) {
+                                 uint32_t stage_info_2, const std::vector<uint32_t> &instrumented_spirv) {
     switch (stage_id) {
         case glsl::kHeaderStageIdMultiEntryPoint: {
             ss << "Stage has multiple OpEntryPoint (";
             bool first_stage = true;
-            for (const auto &insn : instructions) {
-                if (insn.Opcode() == spv::OpFunction) break;  // early exit when possible
-                if (insn.Opcode() == spv::OpEntryPoint) {
+
+            uint32_t offset = ::spirv::kModuleStartingOffset;
+            while (offset < instrumented_spirv.size()) {
+                const uint32_t instruction = instrumented_spirv[offset];
+                const uint32_t length = ::spirv::Length(instruction);
+                const uint32_t opcode = ::spirv::Opcode(instruction);
+
+                // if here, seen all OpString and can return early
```

Comment:
Will fix (and run CI again) after confirm this is actually the way forward

```suggestion
                // if here, seen all OpEntryPoint and can return early
```